libs/accounts/recovery-phone/src/lib/sms.manager.ts

```diff
@@ -73,12 +85,31 @@ export class SmsManager {
       });
       return msg;
     } catch (err) {
+      if (err.code === TwilioErrorCodes.SMS_SEND_RATE_LIMIT_EXCEEDED) {
+        if (retryCount < this.config.maxRetries) {
+          // Exp back off and try again
+          await new Promise((r) =>
+            setTimeout(r, Math.pow(2, retryCount++) * 1000)
+          );
+          return await this._sendSMS(to, body, uid, retryCount);
+        }
+      }
+
       this.metrics.increment('sms.send.error');
 
       if (err.code === TwilioErrorCodes.INVALID_TO_PHONE_NUMBER) {
         throw new RecoveryNumberInvalidFormatError(uid || '', to, err);
       }
+      if (err.code === TwilioErrorCodes.SMS_SEND_RATE_LIMIT_EXCEEDED) {
```

Comment:
Could this moved up as an `else` statement on line 95?

Reply:
That would end up missing the sms.send.error metric.